File: homework/week2/linear_queue.py
MAX_SIZE = 100
class queue:
    list = [MAX_SIZE]
    front = 0
    rear = 0

def isFull(queue):
    return queue.rear == MAX_SIZE

def isEmpty(queue):
    return queue.front == queue.rear

def enqueue(queue, element):
    if isFull:
        return
    queue.list[rear] = element
    rear += 1

def dequeue(queue):
    if isEmpty:
        return
    value = queue.list[front]
    front += 1
    return value

File: homework/week2/2025320058-twuel0921/LinearQueue.java
public class LinearQueue<T> {
    private final int MAX_SIZE = 100;
    private int front;
    private int rear;
    private Object[] elemetArray;

    public LinearQueue() {
        elemetArray = new Object[MAX_SIZE];
        front = 0;
        rear = 0;
    }

    public boolean isEmpty() {
        return front == rear;
    }

    public int size() {
        return front-rear;
    }

    public T enqueue(T item) {
        if(rear == MAX_SIZE) {
            throw new ArrayIndexOutOfBoundsException("Full Queue!");
        }
        elemetArray[rear++] = item;
        return item;
    }

    public T dequeue() {
        if(isEmpty()) {
            throw new ArrayIndexOutOfBoundsException("Empty Queue!");
        }
        return (T) elemetArray[front++];
    }
}
